tracer/src/Datadog.Trace/Util/Http/QueryStringManager.cs

```diff
@@ -12,15 +13,30 @@ namespace Datadog.Trace.Util.Http
     internal class QueryStringManager
     {
         private readonly bool _reportQueryString;
+        private readonly int _maxSizeBeforeObfuscation;
         private readonly Lazy<ObfuscatorBase> _obfuscatorLazy;
 
-        internal QueryStringManager(bool reportQueryString, double timeout, string pattern = null, IDatadogLogger logger = null)
+        internal QueryStringManager(bool? reportQueryString, double? timeout, int? maxSizeBeforeObfuscation, string pattern = null, IDatadogLogger logger = null)
         {
-            _reportQueryString = reportQueryString;
-            pattern ??= Tracer.Instance.Settings.ObfuscationQueryStringRegex;
-            _obfuscatorLazy = new(() => ObfuscatorFactory.GetObfuscator(timeout, pattern, logger, reportQueryString));
+            _reportQueryString = reportQueryString ?? true;
+            _maxSizeBeforeObfuscation = maxSizeBeforeObfuscation ?? 200;
+            pattern ??= TracerSettings.DefaultObfuscationQueryStringRegex;
+            _obfuscatorLazy = new(() => ObfuscatorFactory.GetObfuscator(timeout ?? 100, pattern, logger, _reportQueryString));
```

Comment:
isn't the timeout 200?  maybe we could keep these params non nullable and make sure we get default values from the settings instead of having defaults defined here too 🤔